src/hud.cpp

```diff
@@ -158,24 +158,30 @@ void Hud::drawItem(const ItemStack &item, const core::rect<s32>& rect, bool sele
 	}
 
 //NOTE: selectitem = 0 -> no selected; selectitem 1-based
-void Hud::drawItems(v2s32 upperleftpos, s32 itemcount, s32 offset,
-		InventoryList *mainlist, u16 selectitem, u16 direction)
+void Hud::drawItems(v2s32 upperleftpos, s32 itemcount, s32 inv_offset,
+		InventoryList *mainlist, u16 selectitem, u16 direction, v2s32 offset)
 {
 #ifdef HAVE_TOUCHSCREENGUI
-	if ( (g_touchscreengui) && (offset == 0))
+	if ( (g_touchscreengui) && (inv_offset == 0))
 		g_touchscreengui->resetHud();
 #endif
 
 	s32 height  = m_hotbar_imagesize + m_padding * 2;
-	s32 width   = (itemcount - offset) * (m_hotbar_imagesize + m_padding * 2);
+	s32 width   = (itemcount - inv_offset) * (m_hotbar_imagesize + m_padding * 2);
 
 	if (direction == HUD_DIR_TOP_BOTTOM || direction == HUD_DIR_BOTTOM_TOP) {
 		width  = m_hotbar_imagesize + m_padding * 2;
-		height = (itemcount - offset) * (m_hotbar_imagesize + m_padding * 2);
+		height = (itemcount - inv_offset) * (m_hotbar_imagesize + m_padding * 2);
 	}
 
+	offset.X *= g_settings->getFloat("hud_scaling") *
```

Comment:
Requesting the setting? Twice? Best, it should be cached somewhere, retrieving settings with the hashmap isn't fast at all.